test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/TagHelperComponentTagHelperTest.cs

```diff
@@ -56,7 +66,10 @@ public void Init_InvokesComponentsInitInCorrectOrder()
                     processAsyncCallback: null),
             });
 
-            var testTagHelperComponentTagHelper = new TestTagHelperComponentTagHelper(testTagHelperComponentManager, NullLoggerFactory.Instance);
+            var testTagHelperComponentTagHelper = new TestTagHelperComponentTagHelper(testTagHelperComponentManager, NullLoggerFactory.Instance)
+            {
+                ViewContext = CreateViewContext(new DefaultHttpContext())
```

Comment:
Avoid the duplicate boilerplate by initializing `ViewContext` in `TestTagHelperComponentTagHelper`. Reset property to `null` in `Init_GetsTagHelperComponentPropertyActivator_FromRequestServices`.

Reply:
I don't think I should be resetting it to null in `Init_GetsTagHelperComponentPropertyActivator_FromRequestServices` The `ViewContext` of the `TagHelperComponentTagHelper` is shared with the components.

Reply:
That was the only place I saw where `ViewContext` wasn't set inline with the `TestTagHelperComponentTagHelper` constructor invocation. In any case, the important thing is setting `ViewContext` to the value you want before calling `Init`.